### ISSUE TITLE
Same message received twice from the queue

### ISSUE THREAD

**Issue by barata0** [smallrye/smallrye-reactive-messaging#123]
## Description

Long duration tasks that returns a CompletableFuture on the "@incomming" method are beeing received twice (one for each Thread managed by the ForkJoinPool.commonPool-worker-*)

## Git
I created a small project reproducing the problem: 
[https://github.com/barata0/quarkus-amqp-queue-receive-multiple-times](https://github.com/barata0/quarkus-amqp-queue-receive-multiple-times)

## Actual Behaviour
When there are multiple messages on the queue, the same message is caught by 2 different worker threads. This way they are processed twice

## Expected behaviour
Each message must be processed only once

## Configuration
Apache-ActiveMQ version: 5.15.0
Quarkus version: 0.15.0
Quarkus Plugins: 
* quarkus-smallrye-reactive-messaging
* smallrye-reactive-messaging-amqp

## Classes
Main class: Publish a new message every time GET /app is called and process each message from the queue instantiating a Worker for it.
```java
// App.java
@Path("/app")
public class App {

    private static final Logger logger = LoggerFactory.getLogger(App.class);

    private static AtomicInteger ATOMIC_INTEGER = new AtomicInteger();

    @Inject
    @Stream("my-amqp-output")
    private Emitter<String> emitter;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        String s = "Hello " + ATOMIC_INTEGER.incrementAndGet() + "!";
        emitter.send(s);
        return s;
    }
    
    @Incoming("my-amqp-input")
    public CompletionStage<Void> receive(String input) {
        logger.info("received message with payload: " + input);
        return CompletableFuture.runAsync(new DummyWorker(input));
    }
}
```
Here is the DummyWorker. It sleeps for 2 secs to simulate a long duration task and print log messages

```java
// DummyWorker.java
public class DummyWorker implements Runnable {

    private static final Logger logger = LoggerFactory.getLogger(DummyWorker.class);

    private String payload;

    public DummyWorker(String payload) {
        this.payload = payload;
    }

    @Override
    public void run() {
        try {
            logger.info("starting long task " + payload);
            Thread.sleep(1000); // simulate long task
            logger.info("long task finished " + payload);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}
```

**Comment by cescoffier** [comment]
I've reproduced the issue. It's a bug in the vert.x AMQP client.

**Comment by cescoffier** [comment]
Fixed by https://github.com/vert-x3/vertx-amqp-client/issues/17.

**Comment by cescoffier** [comment]
Let's keep this issue open until a new release is available.

**Comment by cescoffier** [comment]
Fixed in master.

**Comment by HasseNasse** [comment]
I'm able to reproduce the duplicate message issue in my project https://github.com/HasseNasse/NotificationService.

Some information:
- `smallrye-reactive-streams-operators-project:1.0.10`
- `smallrye-reactive-messaging-amqp:1.0.8`
- Application Server is OpenLiberty v19.0.0.11, also tried on 19.0.0.9

I am seeing 2 vertx producers in the AMQ dashboard and whenever a consumer `@Incoming` is attached, I can see 2 consumers being created.

**Comment by cescoffier** [comment]
@HasseNasse your reproducer is using Kafka, can you open a separate issue?
